src/pal/src/misc/cgroup.cpp

```diff
@@ -94,21 +101,156 @@ class CGroup
                 *val = (size_t)temp;
             }
         }
-        free(mem_usage_filename);
+        PAL_free(mem_usage_filename);
         return result;
     }
 
     static bool GetCpuLimit(UINT *val)
+    {
+        if (GetCGroup1CpuLimit(val))
+        {
+            return true;
+        }
+
+        if (GetCGroup2CpuLimit(val))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+private:
+    static bool IsCGroup1MemorySubsystem(const char *strTok){
+        return strcmp("memory", strTok) == 0;
+    }
+
+    static bool IsCGroup1CpuSubsystem(const char *strTok){
+        return strcmp("cpu", strTok) == 0;
+    }
+
+    static bool GetCGroup2CpuLimit(UINT *val)
```

Comment:
This is very similar to `GetCGroup1CpuLimit()`, can't you use the same function for both, maybe passing a parameter, or reading  `s_cgroup_version` inside?